staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -154,6 +155,43 @@ func allowsHTTP2(t *http.Transport) bool {
 	return false
 }
 
+// configureHttp2Transport explicitly configures a net/http HTTP/1 Transport
+// to use golang.org/x/net HTTP/2 instead of bundled HTTP/2.
+// It gives us an opportunity to configure the underlying HTTP/2 Transport.
+// It will return error if the HTTP/1 Transport has already been HTTP/2-enabled.
+func configureHttp2Transport(t1 *http.Transport) error {
+	h2t, err := http2.ConfigureTransports(t1)
+	if err != nil {
+		return err
+	}
+
+	// There are several confirmed flaws with Go's http2 implementation.
+	// So we need to explicitly enable the "http2: connection health check" introduced in golang/net/pull/55
+	// to recover from such "fatal" situations.
+	// Upstream golang issues:
+	//   # golang/go/issues/39750
+	//   # golang/go/issues/40201
+	//   # golang/go/issues/41721
+	// Associated Kubernetes issues:
+	//   # kubernetes/kubernetes/issues/87615
+	if idleConnTimeout := t1.IdleConnTimeout / 2; idleConnTimeout > 0 {
+		// h2Transport configured here does not dial any new connection,
+		// we can set the interval of healthCheck to half of h1Transport's IdleConnTimeout.
+		// By default, ReadIdleTimeout is not configured.
+		h2t.ReadIdleTimeout = idleConnTimeout
+	} else {
+		// The default IdleConnTimeout for h1Transport is 90s
+		h2t.ReadIdleTimeout = defaultTransport.IdleConnTimeout / 2
+	}
+	// The default PingTimeout is 15s.
+	// Our configuration should not below default value.
```

Comment:
Nits: I think the comment is missing some word here.

```suggestion
	// Our configuration should not be below the default value.
```

Reply:
Thx. Will fix later.